### PR TITLE
refactor(geographic): unify and encapsulate crs.

### PR DESCRIPTION
**Pull request by gchoqueux**
## Description
* unify and encapsulate crs.
* crs string aren't hard coded.
* there are two types of crs : `EPSG:XXXX` and `TMS:XXXX`

### CODE REVIEW

File: src/Core/Geographic/Crs.js [hunk 1]
```diff
@@ -2,38 +2,43 @@ import proj4 from 'proj4';
...
+const root_tms = 'TMS';
+const root_epsg = 'EPSG';
+const nCrs = ['4326', '3857'];
 
-const EPSG = [
-    'EPSG:4326',
-    'EPSG:3857',
-];
+const TMS = nCrs.map(n => `${root_tms}:${n}`);
+const EPSG = nCrs.map(n => `${root_epsg}:${n}`);
```
Comment (zarov) [inline src/Core/Geographic/Crs.js]: Too complicated imho, I understand what you are aiming for, but I see no problem in using already completed array. Same goes for `root_`, use directly the string: it won't change any time soon, so no worry about maintainability.

File: src/Core/Geographic/Crs.js [hunk 2]
```diff
@@ -2,38 +2,43 @@ import proj4 from 'proj4';
...
+
+function isTms(crs) {
+    return crs.includes(root_tms);
+}
+
+function isEpsg(crs) {
+    return crs.includes('EPSG');
```
Comment (zarov) [inline src/Core/Geographic/Crs.js]: See, this is not coherent at all, following your initial idea, you should use `root_epsg`.

File: src/Core/Geographic/Crs.js [hunk 3]
```diff
@@ -162,10 +167,14 @@ export default {
...
-     * format crs to tile matrix set notation : WMTS:XXXX.
+     * format crs to tile matrix set notation : TMS:XXXX.
      *
      * @param      {string}  crs     The crs to format
      * @return     {string}  formated crs
      */
     formatToTms,
+    isTms,
+    isEpsg,
+    tms_3857: formatToTms('EPSG:3857'),
```
Comment (zarov) [inline src/Core/Geographic/Crs.js]: Same here, why don't you use `EPSG[1]` ? Even more, you should use `TMS[1]`.

**Review by zarov — CHANGES_REQUESTED** [review]
I feel like the modification done in `Crs.js` adds a lot of confusion on the different operations. Please fix that by simply using strings like `EPSG:4326` or `TMS:3857`, not complicated structure like you are proposing.

File: src/Core/Geographic/Crs.js (hunk 1 above)
Comment (gchoqueux) [reply src/Core/Geographic/Crs.js]: ok, I removed `root_`

**Comment by gchoqueux** [comment]
I simplified the code too

**Review by zarov — APPROVED** [review]
It is way better, thanks for this !

**Pull request merged**